### PR TITLE
Add Proposal C

### PR DESCRIPTION
**Pull request by nishakm**
Signed-off-by: nisha <nisha@ctlfsh.tech>

### CODE REVIEW

**Review by jdolitsky — CHANGES_REQUESTED** [review]
Awesome! Can you add the short description to the README?

File: docs/proposals/PROPOSAL_C.md
Comment (jdolitsky) [inline docs/proposals/PROPOSAL_C.md:20]: Perhaps just drop this line.. There are indeed new endpoints introduced in the registry section, but reader can go there to learn more

Comment (jdolitsky) [reply docs/proposals/PROPOSAL_C.md:20]: or rather, place it under "JSON Schema"

File: docs/proposals/PROPOSAL_C.md
Comment (sudo-bmitch) [inline docs/proposals/PROPOSAL_C.md:96]: Is it possible for a reference to point to a digest that exists in the above object list, or one of the children of those objects? And if so, how should registries handle this with GC policies (there's could be a reference loop)?

File: docs/proposals/PROPOSAL_C.md
Comment (sudo-bmitch) [inline docs/proposals/PROPOSAL_C.md:258]: Do we need to use the objects API, or could the existing blob and manifest APIs be used? If the latter, how would clients know which of the two sets of APIs to use? I'm thinking of this from a client that wants to copy images between registries, or some other higher level activity,  but that doesn't know the new media types.

**Review by sudo-bmitch — APPROVED** [review]
I've left some question inline that I'll have when we review. Those don't need to be answered before merging this.

File: docs/proposals/PROPOSAL_C.md
Comment (imjasonh) [inline docs/proposals/PROPOSAL_C.md:27]: How do v3 objects relate to v2 objects, especially v2 objects which already exist in the registry?

Are all existing OCI images in a registry also available as v3 Nodes that I can attach to, or do I have to push them as Nodes separately to be able to attach them? This is especially relevant for blobs -- we definitely don't want clients to have to push blobs twice, or for registries to have to store them twice.

**Review by jdolitsky — APPROVED** [review]
The formatting looks good. The questions raised above can probably be addressed when we review this.

Nisha - your call to merge whenever youre ready to do so

**Review by imjasonh — APPROVED** [review]
+1 to merging, we can discuss points either here in the PR, or in a future WG meeting.

Thanks for adding this Nisha!

**Pull request merged**
